programs/create-genesis/main.cpp

```diff
@@ -54,15 +61,25 @@ struct config_reader {
 
 
 void config_reader::set_program_options(options_description& cli) {
-    cli.add_options()
-        ("genesis-info,g", bpo::value<bfs::path>(&info_file)->default_value("genesis-info.json"),
-            "the location of the genesis info file (absolute path or relative to the current directory).")
-        ("output-file,o", bpo::value<bfs::path>(&out_file)->default_value("cyberway-genesis.dat"),
-            "the file to write generic genesis data to (absolute or relative path).")
-        ("ee-output-directory,e", bpo::value<bfs::path>(&ee_directory)->default_value("events-genesis"),
-            "the directory to write Event-Engine genesis data files to (absolute or relative path).")
-        ("help,h", "Print this help message and exit.")
-        ;
+    cli.add_options() (
+        "genesis-info,g", bpo::value<bfs::path>(&info_file)->default_value("genesis-info.json"),
+        "Genesis info file (absolute or relative path)."
+    ) (
+        "state", bpo::value<bfs::path>(&out_genesis_file)->implicit_value("cyberway-genesis.dat")->default_value(""),
+        "If set, create state genesis. If path set, it is output state genesis file (absolute or relative path)."
+    ) (
+        "ee", bpo::value<bfs::path>(&out_ee_genesis_dir)->implicit_value("event-genesis")->default_value(""),
+        "If set, create EE genesis. If path set, it is output EE genesis dir (absolute or relative path)."
+    ) (
+        "dump,d", bpo::value<bfs::path>(&dump_dir)->default_value("/var/lib/golosd/witness_node_data_dir/operation_dump"),
+        "EE genesis: operation dump dir from Golos (absolute or relative path)."
+    ) (
+        "last-block,l", bpo::value<uint32_t>(&last_block)->default_value(UINT32_MAX),
+        "EE genesis: last block num to read operations from dump."
+    ) (
```

Comment:
It's better to use more informative options name:
* state-output-directory
* ee-output-directory
* operation-dump (or golos-operation-dump)

Reply:
@s-medvedev ee-output-directory is not only directory now. Now it is also switch to build EE genesis or not to.

Reply:
There are 2 different variants:
1. `ee-output-directory`
2. `ee`

I think, first variant is more useful. It's easier to remember that ee-genesis generated if `ee-output-directory` passed, then see `ee` in script and try to remember why directory specified in this option.